packages/blade/src/components/Input/OTPInput/_decisions/_decisions.md

```diff
@@ -0,0 +1,92 @@
+- [OTP SMS auto-read feature](#otp-sms-auto-read-feature)
+  - [Web](#web)
+  - [Native Android](#native-android)
+  - [Native iOS](#native-ios)
+
+## OTPInput API
+
+<img src="./otp.png" width="300" alt="otp input" />
+
+### Sample Usage
+
+```tsx
+<OTPInput
+  otpLength={4}
+  label="Enter One Time Code"
+  labelPosition="left"
+  helpText="Enter the OTP sent to your phone number"
+  errorText="Invalid OTP"
+  successText="Valid OTP"
+  validationState={inputValue.length < 4 ? 'error' : 'success'}
+/>
+```
+
+
+| Prop | Type | Required | Default | Description |
+|---|---|---|---|---|
+| otpLength | `4`, `6`  | No | `4` | Determines the length of the OTP input. |
+| label | `string` | Yes |  | Determines the label of the input field. Also used as `aria-label` |
+| labelPosition | `top`, `left` | No | `top` | Used to identify if the label of the input field will be placed on the top of the input field or left. `left` is only available for large screen devices. When position is `left` then the width of the label is fixed to `120px` and the content is aligned left. |
+| necessityIndicator | `optional`, `required` | No |  | Indicator to be shown next to the label for emphasizing the necessity of the field. if `optional` is provided then it'll show `${label} (optional)` and if `required` is provided then it'll show `${label} *` |
+| name | `string` | No |  | The name of the input, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name) |
+| value | `string` | No |  | Makes input field [controlled](https://reactjs.org/docs/forms.html#controlled-components) |
+| defaultValue | `string` | No |  | Used to set the default value of input field when it's [uncontrolled](https://reactjs.org/docs/uncontrolled-components.html#default-values) controlled |
+| validationState | `error` , `success`, `none` | No |  | Applies the style to the input field based on validation status and shows either `errorText` or `successText` respectively |
+| helpText | `string` | No |  | Shown when we want to add some hint to the input field. Displayed under the input field. Only one of `helpText`, `errorText` or `successText` is shown at a time in the priority order as `errorText`, `successText`, `helpText` |
+| errorText | `string` | No |  | Shown when the `validationState` of the input field is set to Error. Only one of `helpText`, `errorText` or `successText` is shown at a time in the priority order as `errorText`, `successText`, `helpText` |
+| successText | `string` | No |  | Shown when the `validationState` of the input field is set to Success. Only one of `helpText`, `errorText` or `successText` is shown at a time in the priority order as `errorText`, `successText`, `helpText` |
+| onChange | `({ name, value }) => {}` | No |  | Function called when the value of the input field changes |
+| onBlur | `({ name, value }) => {}` | No |  | Function called when the input field loses focus |
```

Comment:
I was of the assumption that the OTP Input field is a combination of either 4 or 6 separate inputs. How is the `onBlur` supposed to work here? Will it get triggered when each of the inputs lose focus?

Reply:
Yes, `onBlur` of any of the OTP field we should send back the value of the entered OTP